spec/controllers/documents/w2s_controller_spec.rb

```diff
@@ -16,116 +16,5 @@
-      end
-    end
-
-    context "with a non-image document" do
-      let(:document_path) { Rails.root.join("spec", "fixtures", "attachments", "document_bundle.pdf") }
-
-      it "renders the thumbnails" do
-        w2_doc = create :document, :with_upload, document_type: "W-2", intake: intake,
-          upload_path: document_path
-
-        expect { get :edit }.not_to raise_error
-        expect(response.body).to include('document_bundle.pdf')
-      end
-    end
-  end
-
-  describe "#update" do
-    context "with valid params" do
-      let(:valid_params) do
-        {
-          document_type_upload_form: {
-            document: fixture_file_upload("attachments/test-pattern.png", "image/png")
-          }
-        }
-      end
-
-      before do
-        allow(subject).to receive(:send_mixpanel_event).and_return(true)
-      end
-
-      it "appends the W-2 documents to the intake and redirects to :edit" do
-        expect{
-          post :update, params: valid_params
-        }.to change(intake.documents, :count).by 1
-
-        latest_doc = intake.documents.last
-        expect(latest_doc.document_type).to eq "W-2"
-        expect(latest_doc.upload.filename).to eq "test-pattern.png"
-
-        expect(response).to redirect_to w2s_documents_path
-      end
-
-      it "sends the document type to mixpanel" do
-        post :update, params: valid_params
-
-        expect(subject).to have_received(:send_mixpanel_event).with(
-          event_name: "document_uploaded",
-          data: {
-            document_type: "W-2",
-            file_extension: ".png",
-            file_content_type: "image/png"
-          })
-      end
-    end
-
-    context "with a nil document" do
-      let(:params) do
-        {
-          document_type_upload_form: {}
-        }
-      end
-
-      it "redirects back to the W2s page" do
-        expect do
-          post :update, params: params
-        end.not_to raise_error
-        expect(response).to redirect_to w2s_documents_path
-      end
-    end
-
-    context "with invalid params" do
-      let(:invalid_params) do
-        {
-          document_type_upload_form: {
-            document: fixture_file_upload("attachments/test-pattern.html")
-          }
-        }
-      end
-
-      it "does not upload the attachment, redirects to :edit and shows a validation error" do
-        expect {
-          post :update, params: invalid_params
-        }.not_to change(intake.documents, :count)
-
-        expect(response.body).to include I18n.t("validators.file_type")
-        expect(response).to render_template(:edit)
-      end
-
-      it "sends the document type to mixpanel" do
-        post :update, params: invalid_params
-
-        expect(MixpanelService).to have_received(:send_event).with(hash_including(
-          event_name: "validation_error",
-          data: hash_including({
-            invalid_document: true
-          })
-        ))
-      end
-    end
-  end
```

Comment:
the W2 form is never shown anymore, but all of these tests were left in place. i deleted rather than updating.

Reply:
These were meant to test basic functionality that is shared across document pages by putting them on one controller in particular. The specs should have been moved when the W-2 controller was deprecated. If possible, it would be helpful to move these specs to a different document controller, and perhaps add a comment that they are meant to test the behavior of a given document controller.

Reply:
ah gotcha. i believe w-2s were refactored into the new Employment screen, so that may be a natural place to add this back. thanks!

Reply:
turns out this was already done, the specs in question have already been translated to the new employment_controller_spec, they were just left in w2s_controller_spec too. so i think that part is ok to ship.